src/app.rs

```diff
@@ -27,14 +30,16 @@ use crate::server::error::Error as ServerError;
 use crate::utils::retry_tx;
 
 pub struct App {
-    pub database:         Arc<Database>,
-    pub identity_manager: Arc<IdentityManager>,
-    tree_state:           OnceLock<TreeState>,
-    pub config:           Config,
+    pub database:            Arc<Database>,
+    pub identity_manager:    Arc<IdentityManager>,
+    pub transaction_manager: Arc<dyn IdentityTransactionManager + Send + Sync>,
```

Comment:
We could probably make `IdentityTransactionManager: Send + Sync` (`+ 'static` even?) then only leave `Arc<dyn IdentityTransactionManager>` here

Reply:
Thanks!